near-sdk/src/json_types/account.rs

```diff
@@ -1,21 +1,31 @@
 use borsh::{BorshDeserialize, BorshSerialize};
-use serde::Serialize;
-use std::convert::{TryFrom, TryInto};
-use std::fmt;
+use serde::{de, Deserialize, Serialize};
+use std::{borrow::Cow, fmt};
+use std::{convert::TryFrom, str::FromStr};
 
 use crate::env::is_valid_account_id;
 use crate::AccountId;
 
-/// Helper class to validate account ID during serialization and deserializiation
+// TODO: this should probably be a specific error type instead of a string.
+const INVALID_ACCOUNT_ID_MSG: &str = "The account ID is invalid";
+
+/// Helper class to validate account ID during serialization and deserializiation.
+/// This type wraps an [`AccountId`].
+///
+/// # Example
+/// ```
+/// use near_sdk::AccountId;
+/// use near_sdk::json_types::ValidAccountId;
+///
+/// let id: AccountId = "bob.near".to_string();
+/// let validated: ValidAccountId = id.parse().unwrap();
+/// ```
 #[derive(
     Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshDeserialize, BorshSerialize, Serialize,
 )]
 pub struct ValidAccountId(AccountId);
 
 impl ValidAccountId {
-    fn is_valid(&self) -> bool {
-        is_valid_account_id(&self.0.as_bytes())
-    }
     pub fn to_string(&self) -> String {
```

Comment:
Also, this function does nothing, since `to_string()` comes with `ToString` which is auto impl from `fmt::Display`. I kept it in though because technically it's a breaking change although I don't know if you can actually trigger it because `ValidAccountId::to_string(&id)` still works.

Reply:
Yeah, we should remove this once we are ready to to a semver-breaking change.